File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.py
import sys
read_line = sys.stdin.readline

N = int(read_line())
cur = 666  # 제일 작은 어둠의 수
n_doom = 0 # 몇 번째 어둠의 수인지 체크

def is_doom(cur):
    return '666' in str(cur) # 문자열 안에 666이 포함되어 있는지 확인

while True:
    if is_doom(cur):
        n_doom += 1
    
    if n_doom == N:
        print(cur)
        break
    
    cur += 1

File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        sc.close();
        
        int cur = 666; // 제일 작은 어둠의 수
        int nDoom = 0; // 몇 번째 어둠의 수인지 체크
        
        while(true) {
            if (isDoom(cur)) {
                nDoom += 1;
            }
            if (nDoom == N) {
                System.out.println(cur);
                break;
            }
            cur += 1;
        }
    }
    
    private static boolean isDoom(int num) {
        return String.valueOf(num).contains("666");
    }
}